### PR TITLE
sync with develop 

### PR DESCRIPTION
**Pull request by jnadroj**

### CODE REVIEW

**Review by geamdev — APPROVED** [review]
LGTM 💯

File: src/generators/subcomponent.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}: IOptions) => {
+  let componentsTemplatePath;
+  const extension = isOnlyJs ? "js" : "ts";
+  if (isOnlyJs)
+    componentsTemplatePath = `${PATH.SUBCOMPONENT_TEMPLATE}/javascript/button.hbs`;
+  else
+    componentsTemplatePath = `${PATH.SUBCOMPONENT_TEMPLATE}/typescript/button.hbs`;
+
+  const componentsPath = `${componentPath}/components`;
+  const componentsFolderPath = `${componentsPath}/Button`;
```
Comment (joelgurumendi) [inline src/generators/subcomponent.ts:23]: @TitoWin94  Change folder name Button to Example

File: src/generators/subcomponent.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  const componentsTemplate = readFileSync(componentsTemplatePath, "utf-8");
+  const componentsTemplateContent = compile(componentsTemplate)({});
+  const componentsIndexTemplate = readFileSync(
+    `${PATH.SUBCOMPONENT_TEMPLATE}/barrel.hbs`,
+    "utf-8"
+  );
+  const componentsIndexTemplateContent = compile(componentsIndexTemplate)({
+    componentName: "Button",
```
Comment (joelgurumendi) [inline src/generators/subcomponent.ts:32]: @TitoWin94  Change ComponentName Button to Example

File: src/generators/subcomponent.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  });
+  const componentsIndexAsTemplate = readFileSync(
+    `${PATH.SUBCOMPONENT_TEMPLATE}/barrel.hbs`,
+    "utf-8"
+  );
+  const componentsIndexAsTemplateContent = compile(componentsIndexAsTemplate)({
+    componentName: "Button",
```
Comment (joelgurumendi) [inline src/generators/subcomponent.ts:39]: @TitoWin94 Change ComponentName Button to Example

File: src/generators/subcomponent.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  //barrel without alias
+  writeFileSync(
+    `${componentsFolderPath}/index.${extension}`,
+    componentsIndexTemplateContent
+  );
+
+  //subcomponent
+  writeFileSync(
+    `${componentsFolderPath}/Button.${extension}x`,
```
Comment (joelgurumendi) [inline src/generators/subcomponent.ts:61]: @TitoWin94 Change FileName Button to Example

**Review by joelgurumendi — CHANGES_REQUESTED** [review]
@TitoWin94 revisemos los comentarios

**Review by joelgurumendi — CHANGES_REQUESTED** [review]
@TitoWin94 revisemos los comentarios

**Review by joelgurumendi — APPROVED** [review]
LGTM 😄

**Pull request merged**